src/core/utils/compatibility.js

```diff
@@ -101,6 +102,11 @@ export function isCompatibleWithUserAgent({
 
     if (
       addon.type === ADDON_TYPE_OPENSEARCH &&
+      // If window isn't defined but the browser is Firefox, this is likely a
+      // server-side render, so we shouldn't condemn this search plugin to
+      // incompatibility yet, or it will stay disabled.
+      // See: https://github.com/mozilla/addons-frontend/issues/4047
+      _hasWindow &&
```

Comment:
This will probably work but it's a bit hacky (checking for `window` is not the real intent here). I think it would be better to use `config.get('server')` to check if we're rendering on the server or not.

Reply:
Oh right, good call.